From c3e63d4e29e2fff5ef1e297c37802cd5517696fd Mon Sep 17 00:00:00 2001
From: Dominik Poggel <pog@iesy.com>
Date: Tue, 18 Oct 2022 11:36:33 +0200
Subject: [PATCH 1/1] arm64: dts: iesy: set drive strength

GPIO_BB138 not in use.
SPI1 no drive strength set. Changed GPIO-Pin on SoC. Originally intended pin used for USB_PWR_EN
---
 .../arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts | 170 +++++++++++++++++-
 1 file changed, 165 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts b/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts
index aee81b2a21a1..c1884f3a7665 100644
--- a/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts
+++ b/arch/arm64/boot/dts/iesy/iesy-rpx30-eva-mi.dts
@@ -883,12 +883,14 @@
 		phy-supply = <&vcc3v0_pmu>;
 		vbus-supply = <&usb_b_vbus_regulator>;
 		status = "okay";
+		pinctrl-0 = <&u2phy_host_pin>;
 	};
 
 	u2phy_otg: otg-port {
 		phy-supply = <&vcc3v0_pmu>;
 		vbus-supply = <&usb_a_vbus_regulator>;
 		status = "okay";
+		pinctrl-0 = <&u2phy_otg_pin>;
 	};
 };
 
@@ -1016,23 +1018,31 @@
 	i2c0 {
 		i2c0_xfer: i2c0-xfer {
 			rockchip,pins =
-				<0 RK_PB0 1 &pcfg_pull_none_8ma>,
-				<0 RK_PB1 1 &pcfg_pull_none_8ma>;
+				<0 RK_PB0 RK_FUNC_1 &pcfg_pull_none_8ma>,
+				<0 RK_PB1 RK_FUNC_1 &pcfg_pull_none_8ma>;
 		};
 	};
 
 	i2c1 {
 		i2c1_xfer: i2c1-xfer {
 			rockchip,pins =
-				<0 RK_PC2 1 &pcfg_pull_none_8ma>,
-				<0 RK_PC3 1 &pcfg_pull_none_8ma>;
+				<0 RK_PC2 RK_FUNC_1 &pcfg_pull_none_8ma>,
+				<0 RK_PC3 RK_FUNC_1 &pcfg_pull_none_8ma>;
+		};
+	};
+
+	i2c2 {
+		i2c2_xfer: i2c2-xfer {
+			rockchip,pins =
+				<2 RK_PB7 RK_FUNC_2 &pcfg_pull_none_8ma>,
+				<2 RK_PC0 RK_FUNC_2 &pcfg_pull_none_8ma>;
 		};
 	};
 
 	lt8912 {
 		pinctrl_lt8912: pinctrl-lt8912 {
 			rockchip,pins = <1 RK_PC4 RK_FUNC_GPIO &pcfg_pull_none
-							 1 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none>;
+							 1 RK_PC5 RK_FUNC_GPIO &pcfg_pull_none_4ma>;
 		};
 	};
 
@@ -1042,6 +1052,156 @@
 							 3 RK_PD1 RK_FUNC_GPIO &pcfg_pull_up>;
 		};
 	};
+	
+	uart0 {
+		uart0_xfer: uart0-xfer {
+			rockchip,pins =
+				<0 RK_PB2 RK_FUNC_1 &pcfg_pull_none_2ma>;
+		};
+
+		uart0_rts: uart0-rts {
+			rockchip,pins =
+				<0 RK_PB5 RK_FUNC_1 &pcfg_pull_none_2ma>;
+		};
+	};
+
+	uart1 {
+		uart1_xfer: uart1-xfer {
+			rockchip,pins =
+				<1 RK_PC1 RK_FUNC_1 &pcfg_pull_none_2ma>;
+		};
+
+		uart1_rts: uart1-rts {
+			rockchip,pins =
+				<1 RK_PC3 RK_FUNC_1 &pcfg_pull_none_2ma>;
+		};
+	};
+
+	uart2-m1 {
+		uart2m1_xfer: uart2m1-xfer {
+			rockchip,pins =
+				<2 RK_PB6 2 &pcfg_pull_up_2ma>;
+		};
+	};
+
+	gmac {
+		rmii_pins: rmii-pins {
+			rockchip,pins =
+				/* mac_txed */
+				<2 RK_PA0 RK_FUNC_2 &pcfg_pull_none_4ma>,
+				/* mac_txd0 */
+				<2 RK_PA2 RK_FUNC_2 &pcfg_pull_none_4ma>,
+				/* mac_txd1 */
+				<2 RK_PA1 RK_FUNC_2 &pcfg_pull_none_4ma>,
+				/* mac_mdio */
+				<2 RK_PA7 RK_FUNC_2 &pcfg_pull_none_4ma>,
+				/* mac_mdc */
+				<2 RK_PB1 RK_FUNC_2 &pcfg_pull_none_4ma>;
+		};
+	};
+
+	gpio_bb138 {
+		pinctrl_gpio_a: pinctrl-gpio-a {
+			rockchip,pins =
+				<1 RK_PC4 RK_FUNC_GPIO &pcfg_pull_none>,
+				<1 RK_PC5 RK_FUNC_GPIO &pcfg_pull_up_4ma>,
+				<1 RK_PC6 RK_FUNC_GPIO &pcfg_pull_up_4ma>,
+				<1 RK_PC7 RK_FUNC_GPIO &pcfg_pull_up_4ma>,
+				<1 RK_PD0 RK_FUNC_GPIO &pcfg_pull_up_4ma>,
+				<1 RK_PD1 RK_FUNC_GPIO &pcfg_pull_up_4ma>,
+				<3 RK_PD0 RK_FUNC_GPIO &pcfg_pull_none>,
+				<3 RK_PD1 RK_FUNC_GPIO &pcfg_pull_none>;
+		};
+
+		pinctrl_gpio_b: pinctrl-gpio-b {
+			rockchip,pins =
+				<3 RK_PA0 RK_FUNC_GPIO &pcfg_pull_none_2ma>,
+				<3 RK_PA1 RK_FUNC_GPIO &pcfg_pull_none_2ma>,
+				<3 RK_PA2 RK_FUNC_GPIO &pcfg_pull_up_2ma>,
+				<3 RK_PA3 RK_FUNC_GPIO &pcfg_pull_up_2ma>,
+				<3 RK_PA4 RK_FUNC_GPIO &pcfg_pull_up_2ma>,
+				<3 RK_PA5 RK_FUNC_GPIO &pcfg_pull_up_2ma>,
+				<3 RK_PA6 RK_FUNC_GPIO &pcfg_pull_up_2ma>,
+				<3 RK_PA7 RK_FUNC_GPIO &pcfg_pull_up_2ma>;
+		};
+	};
+
+	sdmmc {
+		sdmmc_cmd: sdmmc-cmd {
+			rockchip,pins =
+				<1 RK_PD7 RK_FUNC_1 &pcfg_pull_up_8ma>;
+		};
+
+		sdmmc_clk: sdmmc-clk {
+			rockchip,pins =
+				<1 RK_PD6 RK_FUNC_1 &pcfg_pull_none_8ma>;
+		};
+
+		sdmmc_bus1: sdmmc-bus1 {
+			rockchip,pins =
+				<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_8ma>;
+			};
+
+		sdmmc_bus4: sdmmc-bus4 {
+			rockchip,pins =
+				<1 RK_PD2 RK_FUNC_1 &pcfg_pull_up_8ma>,
+				<1 RK_PD3 RK_FUNC_1 &pcfg_pull_up_8ma>,
+				<1 RK_PD4 RK_FUNC_1 &pcfg_pull_up_8ma>,
+				<1 RK_PD5 RK_FUNC_1 &pcfg_pull_up_8ma>;
+		};
+	};
+
+	pwm1 {
+		pwm1_pin: pwm1-pin {
+			rockchip,pins =
+				<0 RK_PC0 RK_FUNC_1 &pcfg_pull_none_2ma>;
+		};
+	};
+
+	pwm2 {
+		pwm2_pin: pwm2-pin {
+			rockchip,pins =
+				<2 RK_PB5 RK_FUNC_1 &pcfg_pull_none_2ma>;
+		};
+	};
+
+	pwm3 {
+		pwm3_pin: pwm3-pin {
+			rockchip,pins =
+				<0 RK_PC1 1 &pcfg_pull_none_2ma>;
+		};
+	};
+
+	spi1 {
+		/* spi1.sdo */
+		spi1_mosi: spi1-mosi {
+			rockchip,pins =
+				<3 RK_PB4 RK_FUNC_4 &pcfg_pull_none>;
+		};
+
+		/* spi1.csn PIN E17 used! */
+		spi1_csn: spi1-csn {
+			rockchip,pins =
+				<3 RK_PB0 RK_FUNC_4 &pcfg_pull_none>;
+		};
+
+		spi1_clk: spi1-clk {
+			rockchip,pins =
+				<3 RK_PB7 RK_FUNC_4 &pcfg_pull_none>;
+		};
+	};
+
+	u2phy {
+		u2phy_host_pin: host-port {
+			rockchip,pins =
+				<3 RK_PC3 RK_FUNC_1 &pcfg_pull_none_4ma>;
+		};
+
+		u2phy_otg_pin: otg-port {
+			rockchip,pins =
+				<3 RK_PC0 RK_FUNC_1 &pcfg_pull_none_4ma>;
+		};
+	};
 };
 
 /* DON'T PUT ANYTHING BELOW HERE.  PUT IT ABOVE PINCTRL */
-- 
2.30.2

